server/monitor/src/main/java/org/apache/accumulo/monitor/util/logging/RecentLogs.java

```diff
@@ -38,52 +41,44 @@ public class RecentLogs {
 
   private static final int MAX_LOGS = 50;
 
+  private final Cache<String,DedupedEvent> eventsCache =
+      Caffeine.newBuilder().maximumSize(MAX_LOGS).build();
+  private final ConcurrentMap<String,DedupedEvent> events = eventsCache.asMap();
```

Comment:
Is this a snapshot or merely a wrapped view? Would it make sense to remove this and have the methods that call it operate directly on the eventsCache object, and only call `.asMap()` when they need to?

Reply:
Yeah I could do that to make it simpler. asMap() is a wrapped view and it caches it https://github.com/ben-manes/caffeine/blob/26899178496d72a15cb243e61b157115932f96eb/caffeine/src/main/java/com/github/benmanes/caffeine/cache/LocalAsyncCache.java#L641

So calling asMap() as needed is fine.